### PR TITLE
Add JSON key content for auth

### PR DESCRIPTION
**Pull request by omkar-g**
## Release notes
Added optional field `json_key_file_content` to get Service Account JSON key as an String input.

## What does this PR do?

Add optional field `json_key_file_content` to get Service Account JSON key as an String input for authentication.

This will add one more option for user to pass Service Account JSON key. With this option the value of Service Account JSON key can be passed as String, instead of file path.

## Why is it important/What is the impact to the user?

This change will not force user to deal with file upload for Service Account JSON key and it can be given as String.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files (and/or docker env variables)
- [x] I have added tests that prove my fix is effective or that my feature works

## How to test this PR locally

Use the field `json_key_file_content` field instead of `json_key_file` in `.conf` file of logstash. And run the application outside of GCE.

### CODE REVIEW

**Comment by omkar-g** [comment]
@andsel Can get your help to review this PR?

File: README.md [hunk 1]
```diff
@@ -49,7 +49,9 @@ You have two options for authentication depending on where you run Logstash.
 
 1. If you are running Logstash outside of Google Cloud Platform, then you will
 need to create a Google Cloud Platform Service Account and specify the full
-path to the JSON private key file in your config. You must assign sufficient
+path to the JSON private key file in your config in `json_key_file` attribute.
+If you do not want to upload file and want to accept it as a string, then add the content 
+of JSON private key in `json_key_file_content` attribute. You must assign sufficient
```
Comment (andsel) [inline README.md:54]:
```suggestion
path to the JSON private key file in your config in `json_key_file` setting.
If you don't want to upload the file file and use its content as a string, then add the content 
of JSON private key in `json_key_file_content` setting. You must assign sufficient
```

File: spec/inputs/test.json
Comment (andsel) [inline spec/inputs/test.json:1]: Is this a plausible json private key? I would expect some for of key attribute inside

**Review by andsel — CHANGES_REQUESTED** [review]
Hi @omkar-g thanks a lot for your contribution.
I've left some notes on the PR.
Once we finish this, if you want, would be great if you can mirror these changes also on the `logstash-output-google_pubsub` plugin.

File: README.md (hunk 1 above)
Comment (omkar-g) [reply README.md]: @andsel Thanks for the suggestion I have updated the doc in README and index.asciidoc as per your suggestion.

**Comment by omkar-g** [comment]
@andsel Thanks for you review. I have pushed changes as per your suggestion. Please have a look whenever you have time.

Sure, I can have a look in logstash-output-google_pubsub plugin as well once this is done. I havent worked with output plugin, but hoping should be similar.

File: example.conf-tmpl [hunk 2]
```diff
@@ -32,6 +32,8 @@ input {
         # outside of GCE, you will need to specify the service account's
         # JSON key file below.
         #json_key_file => "/home/erjohnso/pkey.json"
+        # or
+        #json_key_file_content => "{****}"
```
Comment (andsel) [inline example.conf-tmpl:36]: Please put something that's plausible as JSON content, for example (if it's the minimal thing need) like:

```suggestion
        #json_key_file_content => '{"private_key_id": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "private_key": "-----BEGIN PRIVATE KEY-----\nxxxxxxxxxxx\n-----END PRIVATE KEY-----\n",}'
```

File: spec/inputs/google_pubsub_spec.rb [hunk 3]
```diff
@@ -33,5 +34,9 @@
     expect {
       plugin = LogStash::Inputs::GooglePubSub.new(bad3)
     }.to raise_error(LogStash::ConfigurationError)
+    expect {
```
Comment (andsel) [inline spec/inputs/google_pubsub_spec.rb:37]: I would make the statement of error that is tested more explicit. For example separating in its own `it` section like:

```ruby
  it "raise a configuration error if both 'json_key_file' and 'json_key_file_content' are present" do
    expect { 
      plugin = LogStash::Inputs::GooglePubSub.new(bad4)
      plugin.register()
    }.to raise_error(LogStash::ConfigurationError)
  end
```

and maybe rename `bad4` in `file_and_content_config_settings`

**Review by andsel — CHANGES_REQUESTED** [review]
Hi @omkar-g we are really close, please consider my latest suggestions as a readability improvement.

Thank's for contributing!

File: example.conf-tmpl (hunk 2 above)
Comment (omkar-g) [reply example.conf-tmpl]: I have updated it as per your suggestion.

**Comment by omkar-g** [comment]
@andsel Thanks for your feedback. I have updated as per your suggestion including the update in `CHANGELOG.md`. 

Please review it whenever you have time. Thanks again!

**Review by andsel — APPROVED** [review]
LGTM

**Comment by andsel** [comment]
PR merged and version 1.4.0 published:  https://rubygems.org/gems/logstash-input-google_pubsub/versions/1.4.0.

Thanks @omkar-g for your contribution 👍

**Comment by omkar-g** [comment]
Thanks @andsel for reviewing my PR!

**Pull request merged**